### PR TITLE
fix/OB-3238: add regex to match urls with the bucket name in the subdomain

### PR DESCRIPTION
**Pull request by enleguizamon**
Error al buscar imagen de faceid:
'Error: Error match s3 utl\n    at RestUtils.<anonymous> (/Users/eli-digi/Documents/digiventures/digiventures-global-mirror/packages/external-mirror-aws-lambda/node_modules/@digiventures.stack/utils-helper/src/utils/RestUtils.ts:69:13)'

Solución:
Agregar una regex que matchee la url de faceid.

Test:
RestUtils.test.ts

## Summary by CodeRabbit

- **New Features**
	- Enhanced URL handling for S3 by supporting both standard and subdomain formats, improving flexibility in processing various S3 URLs.

- **Bug Fixes**
	- Improved error handling for S3 URL matching, ensuring robust operations in the absence of valid URL formats.

- **Tests**
	- Introduced unit tests for the RestUtils class, increasing test coverage for S3 file retrieval and conversion functionality.

### CODE REVIEW

File: src/utils/RestUtils.ts [hunk 1]
```diff
@@ -64,12 +64,23 @@ export class RestUtils {
...
+    const bucketInSubdomainRegex = /https:\/\/(.*)\.s3\.amazonaws\.com\/(.*)/;
+
+    let matchedUrl = document.match(bucketInSubdomainRegex);
+
+    let bucketName
+    let bucketKey
+    if (matchedUrl) {
+    bucketName = matchedUrl[1];
+    bucketKey = matchedUrl[2];
+  } else {
```
Comment (JBeccaceci) [inline src/utils/RestUtils.ts:76]: Donde se abre este if?

File: tests/utils/RestUtils.test.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    let restUtils: RestUtils;
+
+    beforeAll(() => {
+      restUtils = new RestUtils();
+    });
+  
+    test('should get a file from S3 and return it as a base64 string', async () => {
+      const document = 'https://s3.amazonaws.com/test.digiventures/5fd78a652ab79c000150e35b/IdCardFrontService/66a3e6d3b7b555c6bf3332aa-1722017571512'
+
+      const result = await restUtils.getBase64FileAsStringFromS3(document);
```
Comment (JBeccaceci) [inline tests/utils/RestUtils.test.ts:33]: Aca vas a tener que hacer un spy de getObject, porque asi como esta, va a buscar la url al bucket de S3!

File: tests/utils/RestUtils.test.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  
+      console.log('Base64 Result:', result);
+      assert.ok(result);
+    });
+
+    test('should get a file from faceid S3 bucket and return it as a base64 string', async () => {
+        const document = 'https://faceid-image-bucket-dev.s3.amazonaws.com/39910747/9b1190d9-ca6b-4d6b-ac48-4724b7a9d651.png';
+      
+        const result = await restUtils.getBase64FileAsStringFromS3(document);
```
Comment (JBeccaceci) [inline tests/utils/RestUtils.test.ts]: Lo mismo aca

**Review by JBeccaceci — CHANGES_REQUESTED** [review]

File: src/utils/RestUtils.ts (hunk 1 above)
Comment (enleguizamon) [reply src/utils/RestUtils.ts:76]: Se abre para ver si hubo match con la url de faceid y sino le asigna el matchedUrl a las que matchean con la otra regex

**Review by JBeccaceci — APPROVED** [review]

**Pull request merged**